app/models/group.rb

```diff
@@ -12,22 +12,20 @@ class Group < ApplicationRecord
   private
 
   def start_date_after_end_date
-    if start_date.blank? || end_date.blank?
-      return
-    end
-
-    if end_date <= start_date
-      t(".group.new.start_date_after_end_date_error")
+    if start_date.present? && end_date.present? && (end_date <= start_date)
+      errors.add(
+        :start_date,
+        I18n.t(".groups.new.start_date_after_end_date_error"),
+      )
     end
   end
 
   def start_date_in_the_past
-    if start_date.blank? || end_date.blank?
-      return
-    end
-
-    if start_date < Date.today
-      t(".group.new.start_date_in_the_past_error")
+    if start_date.present? && start_date < Date.today
+      errors.add(
+        :start_date,
+        I18n.t(".groups.new.start_date_in_the_past_error"),
+      )
```

Comment:
These changes are from a different PR. Rebase and merge the other PRs and then rebase master on this branch to remove the changes from this PR